### PR TITLE
Fix some PHP8.1 deprecation warnings occurred while running tests

### PR DESCRIPTION
**Pull request by sgiehl**
### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: LdapInterop/UserAccessAttributeParser.php [hunk 1]
```diff
@@ -359,8 +359,8 @@ private function getNormalizedUrl($url, $isThisPiwikUrl = false)
             $parsed['port'] = 80;
         }
 
-        if (substr(@$parsed['path'], -1) !== '/') {
-            $parsed['path'] = @$parsed['path'] . '/';
+        if (!isset($parsed['path']) || substr(@$parsed['path'], -1) !== '/') {
```
Comment (AltamashShaikh) [inline LdapInterop/UserAccessAttributeParser.php]: Not important
we can remove '@' here since we are checking via isset

**Review by AltamashShaikh — APPROVED** [review]

File: LdapInterop/UserAccessAttributeParser.php (hunk 1 above)
Comment (justinvelluppillai) [reply LdapInterop/UserAccessAttributeParser.php]: It can't be removed because we're executing that block when !isset

Comment (justinvelluppillai) [reply LdapInterop/UserAccessAttributeParser.php]: sorry I see what you mean, can be removed

Comment (sgiehl) [reply LdapInterop/UserAccessAttributeParser.php]:
```suggestion
        if (!isset($parsed['path']) || substr($parsed['path'], -1) !== '/') {
```
